tools/profiles/extensions/lto.json

```diff
@@ -5,6 +5,6 @@
     },
     "GCC_ARM": {
         "common": ["-flto"],
-        "ld": ["-u main"]
+        "ld": ["-u main", "-u __wrap_printf", "-u __wrap_sprintf", "-u __wrap_snprintf", "-u __wrap_vprintf", "-u __wrap_vsprintf", "-u __wrap_vsnprintf", "-u __wrap_fprintf", "-u __wrap_vfprintf"]
```

Comment:
The linker flags, if missing, should be added by the build tools if c_lib is set to minimal-printf.

Reply:
These flags only related to LTO at the moment, I believe. basically it tells the linker to keep those symbols when LTO it turned on. If LTO is not used, then those options won't be useful.  So I guess put them in would be fine

Reply:
Is it okay to have the flags if minimal-printf is not used (i.e. c_lib set to `std`)?

Reply:
> These flags only related to LTO at the moment, I believe. basically it tells the linker to keep those symbols when LTO it turned on. If LTO is not used, then those options won't be useful. So I guess put them in would be fine

There's a lot of uncertainty in your comment there James. Are you sure that this is what is going on? Has this been tested comprehensively?

Reply:
> Is it okay to have the flags if minimal-printf is not used (i.e. c_lib set to `std`)?
Yes, it has been tested locally. it works fine both with "target.printf_lib": "minimal-printf" or "std"